src/gateway/Messages.sol

```diff
@@ -820,6 +817,38 @@ library Messages {
     }
 
     // TODO: should be moved to a util contract
+    function stringToBytes128(string memory source) internal pure returns (bytes memory) {
```

Comment:
I would move these and the bytes32 similar methods to a new `src/util/StringLib.sol`. We should also add some basic unit tests for these then.